tests/functional/test_install_vcs_git.py

```diff
@@ -73,31 +24,22 @@ def test_check_version(script):
     assert not git.check_version(version_pkg_path, ['abc123'])
 
 
-@patch('pip.vcs.git.Git.get_short_refs')
-def test_check_rev_options_should_handle_branch_name(get_refs_mock):
-    get_refs_mock.return_value = {'master': '123456', '0.1': '123456'}
+@patch('pip.vcs.git.Git.get_revision_sha')
+def test_check_rev_options_ref_exists(get_sha_mock):
+    get_sha_mock.return_value = '123456'
     git = Git()
 
     result = git.check_rev_options('master', '.', [])
     assert result == ['123456']
 
 
-@patch('pip.vcs.git.Git.get_short_refs')
-def test_check_rev_options_should_handle_tag_name(get_refs_mock):
-    get_refs_mock.return_value = {'master': '123456', '0.1': '123456'}
-    git = Git()
-
-    result = git.check_rev_options('0.1', '.', [])
-    assert result == ['123456']
-
-
-@patch('pip.vcs.git.Git.get_short_refs')
-def test_check_rev_options_should_handle_ambiguous_commit(get_refs_mock):
-    get_refs_mock.return_value = {'master': '123456', '0.1': '123456'}
+@patch('pip.vcs.git.Git.get_revision_sha')
+def test_check_rev_options_ref_not_found(get_sha_mock):
+    get_sha_mock.return_value = None
     git = Git()
 
-    result = git.check_rev_options('0.1', '.', [])
-    assert result == ['123456'], result
+    result = git.check_rev_options('123456', '.', ['rev-option'])
+    assert result == ['rev-option']
```

Comment:
You've removed ", result" here; likely unintentionally. :P

Reply:
Oh, interesting catch, and thanks for reviewing! Though I think what might have happened there is that the diff isn't lining things up, and I wound up modifying the function above (`test_check_rev_options_should_handle_tag_name()` -- which doesn't have `result` (in line 91) -- and deleting the second one, as opposed to vice versa.

Incidentally, is adding "result" needed with pytest? Here is without result:

    >       assert result == ['rev-option2']
    E       AssertionError: assert ['rev-option'] == ['rev-option2']
    E         At index 0 diff: 'rev-option' != 'rev-option2'
    E         Use -v to get the full diff

And here is with result:

    >       assert result == ['rev-option2'], result
    E       AssertionError: ['rev-option']
    E       assert ['rev-option'] == ['rev-option2']
    E         At index 0 diff: 'rev-option' != 'rev-option2'
    E         Use -v to get the full diff

In both cases, it shows what it's getting compared to...

Reply:
> In both cases, it shows what it's getting compared to...

Yeah, that's cool then.

I've been fiddling with `result`s from `script.run` and those show the output of the command when passed as the message for assert, which is extremely useful. That's why I'm a little specific about this. :P 

> thanks for reviewing!

You're welcome! ^.^